add logic to wait for the startup window to finish before showing information dialogs during startup

Montellese opened the pull request:
This is an attempt at solving the problem that when we try to show a modal dialog during startup to provide the user with some information, the dialog isn't really shown to the user most of the time because by the time the dialog becomes visible, Confluence's `Startup.xml` kicks in and executes a `ReplaceWindow()` call to go to the home window. That `ReplaceWindow()` call changes the active window and basically hides the modal dialog in the background.

What this does is that when we want to show a modal dialog with some information we first wait until the `Startup.xml` has been processed and the actual start screen has been reached. Then we show the dialog.

TBH I don't really like this approach but I'm not sure how else to do it. Should `ReplaceWindow()` (and `ActivateWindow()` for that matter) be blocked while a modal dialog is active instead?

Montellese committed:
add logic to wait for the startup window to finish before showing information dialogs during startup

mkortstiege commented:
I don´t like this either. IMO the  `WindowManager` should always respect modals and not be allowed to override them.

Montellese commented:
@mkortstiege: Agreed but I have no idea in how many cases that would be a change in behaviour.

mkortstiege commented:
Yea, pretty scary change. Is there a forum thread for this issue or is it one that popped up during your media importer work? Nonetheless I think we should try it in master so we get a few testers.

Montellese commented:
It's an issue I came across when I tried to show an OK dialog during startup in case the language update logic wasn't able to find, download or install a language addon matching the user's language setting from pre-Isengard. I then noticed that for the XBMC -> Kodi migration path @Memphiz also added an OK dialog during startup informing the user that his userdata has been migrated. That dialog has the same issue as the one I added.

This needs to be solved for Isengard because otherwise users without a working internet connection will start Kodi Isengard and will see the user interface in English independent of the language they had configured. In that case we need to be able to tell them that they either need to connect to the internet and download the language addon in Kodi or they need to download and install the language addon ZIP manually.

Montellese commented:
We still need a fix for this issue. @xhaggi: You've looked into modal dialogs and their handling in the window manager recently. Do you have any proposals/ideas for the problem at hand?

xhaggi commented:
IMO we should introduce a new GUI message `GUI_APPLICATION_STARTED` or something similar and show the migration and language info dialog if it was triggered. We should send the message in case the startup window is closed or if no startup window exists at the very last in `CApplication::Initialize()`. 

This will also work if we display the login window see 
https://github.com/xbmc/xbmc/pull/6666/files#diff-63246197ac0f0ac89358c1155ad21d5bR1199
https://github.com/xbmc/xbmc/blob/master/xbmc/windows/GUIWindowLoginScreen.cpp#L329

Montellese commented:
See #7150 for a cleaner alternative based on @xhaggi's comment.
